tools/applicationinsights-web-snippet/src/snippet.ts

```diff
@@ -202,11 +204,60 @@ declare var cfg:ISnippetConfig;
             "js2.cdn.applicationinsights.io",
             "js2.cdn.monitor.azure.com",
             "az416426.vo.msecnd.net" // this domain is supported but not recommended
-        ]
-    
-        // Assigning these to local variables allows them to be minified to save space:
-        let targetSrc : string = (aiConfig as any)["url"] || cfg.src
-        if (targetSrc) {
+        ];
+
+        let targetSrc : string = (aiConfig as any)["url"] || cfg.src;
+        var match = targetSrc.match(/^(http[s]?:\/\/.*\/)([\w]+\.)?(\d+(\.\d+){0,2})\.(([\w]+\.){0,2}js)$/);
+        if (match.length === 7 && cfg.sri) {
```

Comment:
If there is no "match" because the URL is something unexpected, I think it will be null -- which will then cause an exception here...

We should check if there was a match and maybe even defer the regex until after we check if they have requested (enabled) the `cfg.sri` so that we don't try to do anything until they actually ask us.